locked-asset/proxy_dex/src/pair_interactions.rs

```diff
@@ -21,32 +39,33 @@ pub struct RemoveLiqudityResultWrapper<M: ManagedTypeApi> {
 pub trait PairInteractionsModule {
     fn call_add_liquidity(
         &self,
-        pair_address: ManagedAddress,
-        first_token_id: TokenIdentifier,
-        first_token_amount_desired: BigUint,
-        first_token_amount_min: BigUint,
-        second_token_id: TokenIdentifier,
-        second_token_amount_desired: BigUint,
-        second_token_amount_min: BigUint,
+        args: CallAddLiqArgs<Self::Api>,
```

Comment:
I know this is way cleaner at a SC level, but isn't it more difficult for the one calling/integrating this endpoint? As he needs to also encode the struct, instead of just providing each element.

Reply:
Indeed, you are right.